tests/integration/test_permissions.py

```diff
@@ -0,0 +1,311 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+from helpers import login_user_via_session
+
+from invenio_access.models import ActionUsers
+from invenio_accounts.models import User
+from invenio_collections.models import Collection
+from invenio_db import db
+from invenio_indexer.api import RecordIndexer
+from invenio_pidstore.models import PersistentIdentifier
+from invenio_records.api import Record
+from invenio_search import current_search_client as es
+
+from inspirehep.modules.pidstore.minters import inspire_recid_minter
+from inspirehep.modules.search.api import LiteratureSearch
+
+
+def _create_and_index_record(record):
+    record = Record.create(record)
+    inspire_recid_minter(record.id, record)
+    # invenio-collections will populate _collections field in record upon
+    # commit
+    db.session.commit()
+
+    # Record needs to be indexed since views fetch records from ES
+    r = RecordIndexer()
+    r.index(record)
+    es.indices.refresh('records-hep')
+
+    return record
+
+
+@pytest.fixture(scope='function')
+def sample_record(app):
+    record = {
+        "$schema": "http://localhost:5000/schemas/records/hep.json",
+        "control_number": "123",
+        "titles": [
+            {
+                "title": "Supersymmetric gauge field theory and string theory"
+            }
+        ],
+        "collections": [
+            {
+                "primary": "HEP"
+            },
+            {
+                "primary": "THESIS"
+            }
+        ]
+    }
+
+    record = _create_and_index_record(record)
+    yield record
+
+    pid = PersistentIdentifier.get('literature', '123')
+    db.session.delete(pid)
+    record.delete(force=True)
+    es.delete(index='records-hep', doc_type='hep', id=pid.object_uuid)
+    db.session.commit()
+
+
+@pytest.fixture(scope='function')
+def restricted_record(app):
+    # Create a restricted collection
```

Comment:
Nit: I think the rest of the code is hammering enough the fact that this is a restricted collection... : )